bin/rubocop-daemon-wrapper

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+CACHE_DIR="$(echo ${PWD:1} | tr '/' '+')"
+TOKEN_PATH="$HOME/.cache/rubocop-daemon/$CACHE_DIR/token"
+PORT_PATH="$HOME/.cache/rubocop-daemon/$CACHE_DIR/port"
+if ! [ -f "$TOKEN_PATH" ]; then
+  rubocop-daemon start
+  if ! [ -f "$TOKEN_PATH" ]; then
```

Comment:
Oh I see! Thanks, that makes sense. Yes we should create the cache files before calling `Process.daemon`.

I've also come up with a better way to write the bash script: I call `set -e` so that bash will exit if anything has a non-zero exit code. So if `rubocop-daemon start` crashes with an error or calls `exit 1`, then the script will exit as well. 

But this still has a race condition, because `cat "$TOKEN_PATH"` could be called before the file is created by the Ruby process. So I will also move `Process.daemon` to the bottom of the `#start` method.

Reply:
Ohh actually I need to put it inside the `#make_server_file` block, because `read_socket` is blocking. Otherwise `rubocop-daemon start` would never exit. 

But then `Cache.pid_running?` returns false inside `Daemon.running?`, because the PID has changed. This is very tricky!

So I need to figure out how to keep the Ruby process running until the new daemon process writes the PID to the file. Or maybe just a loop in the bash script that waits for the file. (On my machine, Ruby wrote the file before the bash script tried to read it.)

Reply:
Ok, so I think I've figured out how `Server#start` works:

* Requires rubocop
* Starts the server, so it's now running on `localhost:port` and ready to accept connections
* Calls `Process.daemon`, so the Ruby process now has a new PID and the script exits

#### In the Ruby daemon

* Creates the token, path, pid files
* Starts reading requests from the server

#### In the bash script

* Calls `cat` to get the token and port, but these files may not exist yet (race condition)
* But it *doesn't* need to the get the PID, which changes after `Process.daemon`

So what I can do is create the token and port files before the script exits, and then create the pid file after `Process.daemon` is called.

Reply:
Ok I've fixed the issue! Sorry about that.